cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -1132,18 +1146,21 @@ public void close() {
   }
 
   // if there was an API exception that was not a 4xx, we can just try again
-  private RetryableProvisionException handleApiException(ApiException e) throws RetryableProvisionException {
+  private RetryableProvisionException handleApiException(ApiException e, ErrorGroup errorGroup)
+          throws RetryableProvisionException {
     if (e.getStatusCode().getCode().getHttpStatusCode() / 100 != 4) {
-      throw new RetryableProvisionException(e);
+      throw new RetryableProvisionException(e, ErrorGroup.DATAPROC,
```

Comment:
RetryableException is used to tell the caller than it can retry the operation, it shouldn't ever propagate all the way up. Does it need an error code?

Reply:
I see many paths where it propagates. Here is one
DataprocProvisioner::CreateCluster -> DataprocClient::fromConf

Even the [retry logic](https://github.com/cdapio/cdap/blob/4a32a7fc5679fe40bf322bbd948efa3b8f04b31f/cdap-common/src/main/java/io/cdap/cdap/common/service/Retries.java#L235-L234) that checks if t is RetryableProvisionException, throws t as it is. I don't see anywhere it gets stripped.

Alternatively, I could wrap any exception with ErrorCodeWrappedException and make it the inner exception for a RetryableProvisionException.